csup/segment.go

```diff
@@ -33,6 +33,9 @@ func (s *Segment) Read(r io.ReaderAt, b []byte) error {
 	if len(b) < int(s.MemLength) {
 		return fmt.Errorf("csup: segment read: %w", io.ErrShortBuffer)
 	}
+	if s.Length == 0 {
+		return nil
+	}
 	b = b[:s.MemLength]
```

Comment:
Nit: Don't insert this between the check and use of b and s.MemLength.
```suggestion
	if s.Length == 0 {
		return nil
	}
	if len(b) < int(s.MemLength) {
		return fmt.Errorf("csup: segment read: %w", io.ErrShortBuffer)
	}
	b = b[:s.MemLength]
```